cranelift/codegen/src/isa/call_conv.rs

```diff
@@ -14,7 +16,16 @@ pub enum CallConv {
     Fast,
     /// Smallest caller code size, not ABI-stable.
     Cold,
-    /// Supports tail calls, not ABI-stable.
+    /// Supports tail calls, not ABI-stable except for exception
+    /// payload registers.
+    ///
+    /// On exception resume, a caller to a `tail`-convention function
+    /// assumes that the exception payload values are in the following
+    /// registers (per platform):
+    /// - x86-64: rax, rdx
+    /// - aarch64: x0, x1
+    /// - riscv64: a0, a1
+    /// - pulley{32,64}: x0, x1
```

Comment:
Since we are only defining the exception registers for the `tail` calling convention, can we also check that `try_call`ing a function of a different calling convention triggers verifier errors? At least for now, we can always loosen this later and document the registers used for other conventions at that point. I double checked the `try_call.clif` verifier filetest and didn't see a case for this there.

Reply:
Done!